addons/website/models/website_menu.py

```diff
@@ -41,6 +41,7 @@ def _set_field_is_mega_menu(self):
     name = fields.Char('Menu', required=True, translate=True)
     url = fields.Char('Url', default='')
     page_id = fields.Many2one('website.page', 'Related Page', ondelete='cascade')
+    controller_page_id = fields.Many2one('website.controller.page', 'Related Model Page', ondelete='cascade')
```

Comment:
This is the only change in `website_menu`, but if your menu is related to a `controller_page_id`, shouldn't you alter the `_compute_visible` method to take that into account? Eg if your controller page is unpublished, the menu should be too